model/project_parser_storage.go

```diff
@@ -77,3 +78,37 @@ func ParserProjectUpsertOne(ctx context.Context, settings *evergreen.Settings, m
 	defer ppStorage.Close(ctx)
 	return ppStorage.UpsertOne(ctx, pp)
 }
+
+// ParserProjectUpsertOneWithS3Fallback attempts to upsert the parser project
+// into persistent storage using the given storage method. If it fails due to
+// DB document size limitations, it will attempt to fall back to using S3
+// to store it. If it succeeds, this returns the actual project storage method
+// used to persist the parser project; otherwise, it returns the
+// originally-requested storage method.
+func ParserProjectUpsertOneWithS3Fallback(ctx context.Context, settings *evergreen.Settings, method evergreen.ParserProjectStorageMethod, pp *ParserProject) (evergreen.ParserProjectStorageMethod, error) {
+	err := ParserProjectUpsertOne(ctx, settings, method, pp)
+	if err == nil {
+		return method, nil
+	}
+	if method == evergreen.ProjectStorageMethodS3 {
```

Comment:
Can we combine these two statements that both return `method, err`?

Reply:
Sure.